framework/meta-lib/src/tools/wasm_extractor.rs

```diff
@@ -1,31 +1,40 @@
 use colored::Colorize;
-use std::fs;
+use std::{
+    collections::{HashMap, HashSet},
+    fs,
+};
 use wasmparser::{
-    BinaryReaderError, DataSectionReader, FunctionBody, ImportSectionReader, Parser, Payload,
+    BinaryReaderError, DataSectionReader, ExportSectionReader, FunctionBody, ImportSectionReader,
+    Operator, Parser, Payload,
 };
 
 use crate::ei::EIVersion;
 
 use super::report_creator::{ReportCreator, WITHOUT_MESSAGE, WITH_MESSAGE};
 
+type CallGraph = HashMap<usize, HashSet<usize>>;
+
 const PANIC_WITH_MESSAGE: &[u8; 16] = b"panic occurred: ";
 const PANIC_WITHOUT_MESSAGE: &[u8; 14] = b"panic occurred";
 const ERROR_FAIL_ALLOCATOR: &[u8; 27] = b"memory allocation forbidden";
 const MEMORY_GROW_OPCODE: u8 = 0x40;
+const WRITE_OP: [&str; 1] = ["mBufferStorageStore"];
```

Comment:
1. &[&str] = &[...]; - static slice is easier to write and cleaner
2. there are more write hooks: storageStore, int64storageStore, bigIntStorageStoreUnsigned, smallIntStorageStoreUnsigned, smallIntStorageStoreSigned
3. we should probably extend this to contract calls & transfers, they shouldn't be performed in a view, I think; event logs are also questionable